Framework/AnalysisSupport/src/AODJAlienReaderHelpers.cxx

```diff
@@ -278,6 +281,19 @@ AlgorithmSpec AODJAlienReaderHelpers::rootFileReaderCallback()
         }
         first = false;
       }
+      int64_t stopSize = totalSizeCompressed;
+      int64_t bytesDelta = stopSize - startSize;
+      int64_t stopTime = uv_hrtime();
+      float currentDelta = float(stopTime - startTime) / 1000000000; // in s
+      if (ceil(maxRate) > 0.) {
+        float extraTime = (bytesDelta / 1000000 - currentDelta * maxRate) / maxRate;
+        // We only sleep if we read faster than the max-read-rate.
+        LOGP(info, "Read {} MB in {} s. Sleeping for {} seconds to stay within {} MB/s limit.", bytesDelta / 1000000, currentDelta, extraTime, maxRate);
```

Comment:
Shouldn't we put the log only when extraTime is actually > 0 ?

Reply:
fair enough.